temporal-testing-junit4/build.gradle

```diff
@@ -0,0 +1,165 @@
+plugins {
+    id 'java-library'
+    id 'net.ltgt.errorprone' version '1.3.0'
+    id 'net.minecrell.licenser' version '0.4.1'
+    id 'com.palantir.git-version' version '0.12.3'
+    id 'maven-publish'
+    id 'signing'
+    id 'de.marcphilipp.nexus-publish' version '0.4.0'
+    id 'name.remal.check-updates' version '1.1.6'
+}
+
+apply plugin: 'maven-publish'
+apply plugin: 'de.marcphilipp.nexus-publish'
+apply plugin: 'java'
+
+if (hasProperty('signing.keyId')) {
+    apply plugin: 'signing'
+    signing {
+        sign configurations.archives
+    }
+}
+
+group = 'io.temporal'
+version = getVersionName()
+archivesBaseName = "temporal-testing-junit4"
+
+description = '''Temporal Workflow Java SDK testing, JUnit 4.x integration'''
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+    withJavadocJar()
+    withSourcesJar()
+}
+
+dependencies {
+    errorproneJavac('com.google.errorprone:javac:9+181-r4173-1')
+    errorprone('com.google.errorprone:error_prone_core:2.4.0')
+
+    api project(':temporal-testing')
+
+    implementation group: 'junit', name: 'junit', version: '4.13.1'
+}
+
+license {
+    header rootProject.file('license-header.txt')
+    exclude '**/*.puml'
+}
+
+compileJava {
+    dependsOn 'googleJavaFormat'
+    options.encoding = 'UTF-8'
+    options.compilerArgs << '-Xlint:none' << '-Xlint:deprecation' << '-Werror'
+}
+
+compileTestJava {
+    options.encoding = 'UTF-8'
+    options.compilerArgs << '-Xlint:none' << '-Xlint:deprecation' << '-Werror'
+}
+
+if (JavaVersion.current().isJava8Compatible()) {
+    allprojects {
+        tasks.withType(Javadoc) {
+            options.addStringOption('Xdoclint:none', '-quiet')
+        }
+    }
+}
+
+javadoc {
+    options.encoding = 'UTF-8'
+    if (JavaVersion.current().isJava9Compatible()) {
+        options.addBooleanOption('html5', true)
+    }
+}
+
+task sourceJar(type: Jar) {
+    from sourceSets.main.allSource
+    classifier "sources"
+}
+
+test {
+    dependsOn 'checkLicenseMain'
+    testLogging {
+        events 'passed', 'skipped', 'failed'
+        exceptionFormat 'full'
+        // Uncomment the following line if you want to see test logs in gradlew run.
+        showStandardStreams true
+    }
+    forkEvery = 1
+    maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1
+}
+
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            from components.java
+            versionMapping {
+                usage('java-api') {
+                    fromResolutionOf('runtimeClasspath')
+                }
+                usage('java-runtime') {
+                    fromResolutionResult()
+                }
+            }
+            pom {
+                name = 'Temporal Java SDK testing'
+                packaging = 'jar'
+                // optionally artifactId can be defined here
+                description = 'Temporal Workflow Java SDK testing'
```

Comment:
Temporal Workflow Java SDK testing, JUnit 4.x integration

Reply:
Updated. I guess the `name` has to be updated as well, I've changed it to `Temporal Java SDK testing for JUnit 4`, not sure if that might be too long for a name though.